server/commands/battle.py

```diff
@@ -0,0 +1,597 @@
+
+
+def ooc_cmd_create_move(client, arg):
+    """
+    Allow you to create a move for a fighter.
+    You have to choose a fighter first!
+    MovesType: Atk or Spa
+    Usage: /create_move MoveName MovesType Power Accuracy Effects
+    """
+    if client.battle is not None:
+        args = arg.split(" ")
+        if args[2].isdigit and float(args[2]) > 0:
+            if args[3].isdigit and float(args[3]) > 0 and float(args[3]) < 101:
+                if args[1].lower() in ["atk", "spa"]:
+                    with open(
+                        "storage/battlesystem/battlesystem.yaml", "r", encoding="utf-8"
+                    ) as c_load:
+                        char = yaml.safe_load(c_load)
+                        fighter = client.battle.fighter
+                        move_list = []
+                        for i in range(0, len(char[fighter]["Moves"])):
+                            move_list.append(char[fighter]["Moves"][i]["Name"])
+                        if args[0].lower() not in move_list:
+                            char[fighter]["Moves"].append({})
+                            index = len(char[fighter]["Moves"]) - 1
+                            char[fighter]["Moves"][index]["Name"] = args[0].lower()
+                            char[fighter]["Moves"][index]["MovesType"] = args[1].lower()
+                            char[fighter]["Moves"][index]["Power"] = float(args[2])
+                            char[fighter]["Moves"][index]["Accuracy"] = float(args[3])
+                            char[fighter]["Moves"][index]["Effects"] = []
+                            for i in range(4, len(args)):
+                                if args[i].lower() in battle_effects:
+                                    char[fighter]["Moves"][index]["Effects"].append(
+                                        args[i].lower()
+                                    )
+                            with open(
+                                "storage/battlesystem/battlesystem.yaml",
+                                "w",
+                                encoding="utf-8",
+                            ) as c_save:
+                                yaml.dump(char, c_save)
+                                client.battle = ClientManager.Battle_char(
+                                    client, fighter
+                                )
+                                client.send_ooc(f"{args[0]} has been added!")
+                        else:
+                            client.send_ooc("This move has already been created.")
+                else:
+                    client.send_ooc("Move's Type should be atk or spa!")
+            else:
+                client.send_ooc("Accuracy should be a integer between 1 and 100")
+        else:
+            client.send_ooc("Power argument is not an integer.")
+    else:
+        client.send_ooc(
+            "You have to choose a figher to create a move.\n /choose_fighter FighterName"
+        )
```

Comment:
if if if if else else else else staircase is very hard to read, please revise

Reply:
Done!